lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -130,10 +130,15 @@ bool ReabstractionInfo::prepareAndCheck(ApplySite Apply, SILFunction *Callee,
   bool HasConcreteGenericParams = false;
   bool HasNonArchetypeGenericParams = false;
   HasUnboundGenericParams = false;
-  for (auto DT : CalleeGenericSig->getGenericParams()) {
+  auto *SM = Apply.getModule().getSwiftModule();
+  for (auto GP : CalleeGenericSig->getGenericParams()) {
```

Comment:
Instead you can call GenericSignature::getSubstitutableParams(), which skips those that are concrete